### PR TITLE
Add new item: Allied Cyclone

### PR DESCRIPTION
**Pull request by VilkkuV**
#121

### CODE REVIEW

File: game/scripts/npc/items/item_allied_cyclone.txt [hunk 1]
```diff
@@ -100,7 +99,63 @@
...
       }
     }
-  }
+      "OnSpellStart"
+    {
+
+      "ApplyModifier"
+      {
+        "Target"                                          "TARGET"
+        "ModifierName"                                    "modifier_item_allied_cyclone_active"
```
Comment (Trildar) [inline game/scripts/npc/items/item_allied_cyclone.txt]: I think you might be able to just apply the in-built modifier. It's called `modifier_eul_cyclone`.

File: game/scripts/npc/items/item_allied_cyclone.txt [hunk 2]
```diff
@@ -100,7 +99,63 @@
...
+        "Duration"                                        "%allied_cyclone_duration"
+      }
+    }
+    "Modifiers"
+    {
+      "modifier_item_allied_cyclone"
+      {
+        "Passive"                                         "1"
+        "IsHidden"                                        "1"
```
Comment (Trildar) [inline game/scripts/npc/items/item_allied_cyclone.txt]: Should have `"Attributes"      "MODIFIER_ATTRIBUTE_MULTIPLE"` and `"IsPurgable"      "0"`. Same goes for all the other levels.

File: game/scripts/npc/items/item_allied_cyclone_2.txt [hunk 3]
```diff
@@ -88,5 +88,63 @@
...
+        "Passive"                                         "1"
+        "IsHidden"                                        "1"
+        "Properties"
+        {
+          "MODIFIER_PROPERTY_STATS_INTELLECT_BONUS"         "%bonus_intellect"
+          "MODIFIER_PROPERTY_MANA_REGEN_PERCENTAGE"         "%bonus_mana_regen"
+          "MODIFIER_PROPERTY_MOVESPEED_BONUS_CONSTANT"      "%bonus_movement_speed"
+        }
+      }
+      "modifier_item_allied_cyclone_2_active"
```
Comment (Trildar) [inline game/scripts/npc/items/item_allied_cyclone_2.txt]: I don't think there's any reason to name the different level modifiers differently.

**Comment by Trildar** [comment]
Seems to work. Lacking sounds though. Not sure whether that's doable in DataDriven. I wouldn't know how, even if it was possible.

**Comment by chrisinajar** [comment]
We have sounds used somewhere in a datadriven, you can even specify precache lists in there. #documentation-help guys can tackle this, they're KV wizards.

Gonna merge as is so we can play with it, will fix sounds later. Thanks Vikku!.. and also thanks Trildar for reviewing for me :+1:

**Comment by Chronophylos** [comment]
Datadriven..
I don't like it. Especially the `RunScript` stuff.

**Comment by allanvbhg** [comment]
What does this item do again?

**Comment by Haganeko** [comment]
It's like Eul's, but you can use it on allies.

**Pull request merged**